### PR TITLE
Use eth price

### PR DESCRIPTION
**Pull request by bachstatter**
Adds a hook for getting the ETH price. This assumes that WETH will always be used as collateral. Which I think is reasnable

### CODE REVIEW

File: v3/lib/useEthPrice/useEthPrice.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { useCollateralTypes } from '@snx-v3/useCollateralTypes';
+import { useCoreProxy } from '@snx-v3/useCoreProxy';
+import { wei } from '@synthetixio/wei';
+import { useQuery } from '@tanstack/react-query';
+
+export const useEthPrice = () => {
+  const { data: collateralTypes } = useCollateralTypes();
```
Comment (jmzwar) [inline v3/lib/useEthPrice/useEthPrice.ts:7]: Can we have consistent casing here

Comment (bachstatter) [reply v3/lib/useEthPrice/useEthPrice.ts:7]: Hmm I usually capital case + camel for contracts. And lowercase camel for data. Or can you give an example of what you meant

Comment (jmzwar) [reply v3/lib/useEthPrice/useEthPrice.ts:7]: Ah ok - they are different. NVM

**Review by jmzwar — APPROVED** [review]

File: v3/lib/useEthPrice/useEthPrice.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  const { data: CoreProxy } = useCoreProxy();
+
+  return useQuery({
+    queryKey: [{ collateralTypes: collateralTypes, CoreProxy: CoreProxy?.address }, 'eth-price'],
```
[the remark this answers is not in the document]
Comment (bachstatter) [reply v3/lib/useEthPrice/useEthPrice.ts:11]: yea make sense

File: v3/lib/useEthPrice/useEthPrice.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { useQuery } from '@tanstack/react-query';
+
+export const useEthPrice = () => {
+  const { data: collateralTypes } = useCollateralTypes();
+  const { data: CoreProxy } = useCoreProxy();
+
+  return useQuery({
+    queryKey: [
+      {
+        collateralTypes: collateralTypes?.map((x) => x.tokenAddress),
```
[the remark this answers is not in the document]
Comment (bachstatter) [reply v3/lib/useEthPrice/useEthPrice.ts:13]: haha this would be a very rare case but sure

**Pull request merged**